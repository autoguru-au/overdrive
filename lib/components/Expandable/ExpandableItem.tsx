import { wrapEvent } from '@autoguru/utilities';
import cx from 'clsx';
import React, {
	FunctionComponent,
	memo,
	useContext,
	useRef,
} from 'react';
import { ChevronDownIcon, Icon } from '../Icon';
import styles from './style.scss';
import { ExpandableContext, IExpandableContext } from './Expandable';

export interface IProps {
	className?: string;
	id?: string;
	open?: boolean;
	title: JSX.Element;
	body?: JSX.Element;

	onClick?(e: React.MouseEvent<HTMLElement, MouseEvent>): void;

	onChange?(open: boolean): void;
}

const expandableClicked = (id, expandableClicked: (id: string) => void) => {
	expandableClicked(id);
};

const ExpandableItemComponent: FunctionComponent<IProps> = ({
	className = '',
	id,
	open = false,
	title,
	body,
	onClick,
	onChange = () => void 0,
}) => {
	console.log('__________________REDDRAW');
	const expandableContext: IExpandableContext = useContext(ExpandableContext);
	const map =
		expandableContext && expandableContext.openedItemsMap
			? expandableContext.openedItemsMap[id]
			: null;
	const isOpen = expandableContext ? map && map.open : open;
	const hasTopGap = map && map.topGap;
	const hasBottomGap = map && map.bottomGap;

	const prevOpen = useRef<boolean>(isOpen);

	if (prevOpen.current !== isOpen) {
		prevOpen.current = isOpen;
		onChange(isOpen);
	}

	return (
		<article
			className={cx(
				styles.expandableItem,
				{
					[styles.open]: isOpen,
					[styles.topGap]: hasTopGap,
					[styles.bottomGap]: hasBottomGap,
				},
				className
			)}
			aria-label="expandable item"
			aria-expanded={isOpen ? 'true' : 'false'}>
			<header
				aria-label="expandable item header"
				className={styles.title}
				onClick={wrapEvent(
					() =>
						expandableContext &&
						typeof expandableContext.expandableClicked ===
							'function' &&
						expandableClicked(
							id,
							expandableContext.expandableClicked
						),
					onClick
				)}>
				<Icon
					className={styles.arrow}
					size={16}
					icon={ChevronDownIcon}
				/>
				{title}
			</header>
			{isOpen && body && (
				<section
					aria-label="expandable item body"
					className={styles.body}>
					{body}
				</section>
			)}
		</article>
	);
};

export const ExpandableItem = memo(ExpandableItemComponent);
