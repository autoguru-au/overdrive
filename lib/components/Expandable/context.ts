import { createContext } from 'react';
import { ListItemMap } from './Expandable';

export interface ExpandableContext {
	openedItemsMap: ListItemMap;
	multi: boolean;

	expandableClicked(id: string): void;
}

export const ExpandableContext = createContext<ExpandableContext>(null);
