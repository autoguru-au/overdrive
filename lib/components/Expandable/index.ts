export {
	ExpandableItem,
	IProps as IExpandableItemProps,
} from './ExpandableItem';
