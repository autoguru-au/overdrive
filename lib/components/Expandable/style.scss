@import '../../theme/tokens/facade';

$titleHeight: 40px;
$iconSize: 16px;
$iconOffsetX: 3px;
$iconOffsetY: 0px;

$itemBleed: var(--global--space--3);
$itemPadding: var(--global--space--3);

.expandable {
	width: 100%;
	flex-direction: column;
	box-sizing: border-box;
	display: flex;
	align-content: stretch;
	justify-content: flex-start;
	align-items: stretch;
	max-width: 100%;
}

.expandable-item {
	margin: 0 calc(-1 * #{$itemBleed});
	padding: 0 $itemBleed;
	z-index: 0;
	border-radius: 4px;
	transition: background-color 0.2s cubic-bezier(0, 0, 0.2, 1) 0s,
		box-shadow 0.2s cubic-bezier(0, 0, 0.2, 1) 0s,
		border-color 0.2s cubic-bezier(0, 0, 0.2, 1) 0s;
	will-change: box-shadow;
	border: solid 1px transparent;

	.title {
		background-color: white;
	}

	&:not(.open) {
		&:nth-child(odd) {
			.title {
				background-color: getColour(gray, 100);

				.open & {
					background-color: white;
				}
			}
		}

		&:nth-child(even) {
			.title {
				background-color: white;
			}
		}
	}

	&.open {
		background-color: white;
		border-color: getColour(gray, 300);
		box-shadow: 0 1px 10px 0 rgba(0, 0, 0, 0.03),
			0 4px 5px 0 rgba(0, 0, 0, 0.03), 0 2px 4px -1px rgba(0, 0, 0, 0.05);
		z-index: 1;

		&.top-gap {
			margin-top: var(--global--space--4);
		}

		&.bottom-gap {
			margin-bottom: var(--global--space--4);
		}
	}
}

.title {
	position: relative;
	min-height: $titleHeight;
	cursor: pointer;
}

.body,
.title {
	padding: var(--global--space--2);
	background-color: white;
	width: 100%;
}

.arrow {
	transition: transform 0.2s cubic-bezier(0, 0, 0.2, 1) 0s;
	will-change: transform;
	position: absolute;
	left: -$iconOffsetX + $itemBleed;
	top: 0.5 * ($titleHeight - $iconSize) + $iconOffsetY;

	.open & {
		transform: rotate(180deg);
	}
}
