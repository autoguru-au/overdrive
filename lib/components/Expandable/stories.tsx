import { action } from '@storybook/addon-actions';
import { boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { Text } from '../Text';
import { Expandable } from './Expandable';
import { ExpandableItem } from './ExpandableItem';

storiesOf('Components|Expandable', module)
	.addDecorator(story => (
		<div
			style={{
				height: '100%',
				width: '100%',
				backgroundColor: 'white',
				flexDirection: 'row',
				boxSizing: 'border-box',
				display: 'flex',
				placeContent: 'center',
				alignItems: 'center',
			}}>
			<div
				style={{
					maxWidth: '300px',
					width: '100%',
					backgroundColor: 'white',
				}}>
				{story()}
			</div>
		</div>
	))
	.add('default list', () => (
		<Expandable multi={boolean('multi', false)}>
			{new Array(5).fill(0).map((_, index) => (
				<ExpandableItem
					key={Math.round(Math.random() * 1e6)}
					title={
						<div style={{ paddingLeft: '24px' }}>
							<Text>Item {index + 1}</Text>
						</div>
					}
					body={<Text>Body {index + 1}</Text>}
				/>
			))}
		</Expandable>
	))
	.add('default solo item', () => (
		<ExpandableItem
			onClick={action('onClick')}
			open={boolean('open', false)}
			title={
				<div style={{ paddingLeft: '24px' }}>
					<Text>Solo Item</Text>
				</div>
			}
			body={<Text>Solo Item Body</Text>}
		/>
	))
	.add('with all items closed', () => (
		<Expandable multi={true}>
			{new Array(5).fill(0).map((_, index) => (
				<ExpandableItem
					key={Math.round(Math.random() * 1e6)}
					open={false}
					title={
						<div style={{ paddingLeft: '24px' }}>
							<Text>Item {index + 1}</Text>
						</div>
					}
					body={<Text>Body {index + 1}</Text>}
				/>
			))}
		</Expandable>
	))
	.add('multi list with all items set to expanded', () => (
		<Expandable multi={true}>
			{new Array(5).fill(0).map((_, index) => (
				<ExpandableItem
					key={Math.round(Math.random() * 1e6)}
					open={true}
					title={
						<div style={{ paddingLeft: '24px' }}>
							<Text>Item {index + 1}</Text>
						</div>
					}
					body={<Text>Body {index + 1}</Text>}
				/>
			))}
		</Expandable>
	))
	.add('none-multi list with all items set to expanded', () => (
		<Expandable multi={false}>
			{new Array(5).fill(0).map((_, index) => (
				<ExpandableItem
					key={Math.round(Math.random() * 1e6)}
					open={true}
					title={
						<div style={{ paddingLeft: '24px' }}>
							<Text>Item {index + 1}</Text>
						</div>
					}
					body={<Text>Body {index + 1}</Text>}
				/>
			))}
		</Expandable>
	));
