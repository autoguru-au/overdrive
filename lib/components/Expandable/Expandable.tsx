import cx from 'clsx';
import React, { Children, FunctionComponent, useState } from 'react';
import styles from './style.scss';
import { ExpandableContext } from './context';

export interface Props {
	className?: string;
	multi?: boolean;
	children: any;

	onChange?(): void;
}

export interface ListItemMap {
	[index: string]: {
		open: boolean;
		topGap?: boolean;
		bottomGap?: boolean;
	};
}

const cloneChildren = (children: any): any[] =>
	children
		? Children.map(children, (expandable, index) => {
				const id = expandable.props.id || expandable.key || index;

				return {
					...React.cloneElement(expandable, {
						...expandable.props,
						id,
						key: id,
					}),
					key: id,
				};
		  })
		: null;

const buildChildrenOpenMap = (
	expandables: any[],
	multi: boolean,
): ListItemMap =>
	expandables
		? expandables.reduce((map, child) => {
				if (
					child.props.id &&
					(multi ||
						!Object.keys(map).reduce(
							(hasOpen, currentKey) =>
								hasOpen ? true : map[currentKey].open,
							false,
						))
				) {
					map[child.props.id] = {
						open: child.props.open,
					};
				}

				return map;
		  }, {})
		: {};

const updateMapGas = (
	ids: string[],
	openedItemsMap: ListItemMap,
): ListItemMap =>
	ids.reduce((map: ListItemMap, currentId, index) => {
		map[currentId] = map[currentId] || { open: false };
		map[currentId].topGap = false;
		map[currentId].bottomGap = false;

		if (index === 0 && openedItemsMap[currentId].open) {
			map[currentId].topGap = false;
			map[currentId].bottomGap = true;
		} else if (openedItemsMap[currentId].open) {
			map[currentId].topGap = !openedItemsMap[ids[index - 1]].open;
			map[currentId].bottomGap = true;
		}

		if (index === ids.length - 1) {
			map[currentId].bottomGap = false;
		}

		return map;
	}, openedItemsMap);

const generateOpenedMapWithGaps = (
	id: string,
	expandables: any[],
	openedItemsMap: ListItemMap,
	multi: boolean,
): ListItemMap => {
	const ids = expandables.map(expandable => expandable.props.id);

	const newOpenMap: ListItemMap = {
		...(multi ? openedItemsMap : {}),
		[id]: {
			open: !(openedItemsMap[id] && openedItemsMap[id].open),
		},
	};

	return updateMapGas(ids, newOpenMap);
};

export const Expandable: FunctionComponent<Props> = ({
	className = '',
	multi = false,
	children,
	onChange = () => void 0,
}) => {
	const expandables: any[] = cloneChildren(children);

	const [openedItemsMap, setOpenedItemsMap] = useState<ListItemMap>(
		// Honor explicitly set to expanded list items considering the multi option
		buildChildrenOpenMap(expandables, multi),
	);

	const expandableClicked = (id: string) => {
		setOpenedItemsMap(
			generateOpenedMapWithGaps(id, expandables, openedItemsMap, multi),
		);
		onChange();
	};

	return (
		<ExpandableContext.Provider
			value={{ openedItemsMap, expandableClicked, multi }}>
			<div
				className={cx(styles.expandable, className)}
				aria-label="expandable list">
				{expandables}
			</div>
		</ExpandableContext.Provider>
	);
};
