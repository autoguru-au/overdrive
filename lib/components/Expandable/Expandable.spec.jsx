import React, { useState } from 'react';
import { mount, shallow } from 'enzyme';
import { Expandable } from './Expandable';
import { act } from 'react-dom/test-utils';
import { ExpandableItem } from './ExpandableItem';

describe.skip('<Expandable />', () => {
	it('should not throw with no items', () =>
		expect(() => shallow(<Expandable />)).not.toThrow());

	it('should not throw with default item', () =>
		expect(() =>
			shallow(
				<Expandable>
					<ExpandableItem />
				</Expandable>,
			),
		).not.toThrow());

	it('should match snapshot for list without items', () => {
		let testComponent;
		act(() => {
			const TestComponent = () => {
				return <Expandable />;
			};

			testComponent = mount(<TestComponent />);
		});

		expect(testComponent.html()).toMatchSnapshot();
	});

	it('should match snapshot for list with a closed item', () => {
		let testComponent;
		act(() => {
			const TestComponent = () => {
				return (
					<Expandable>
						<ExpandableItem
							title={<h1>Title</h1>}
							body={<p>Body</p>}
						/>
					</Expandable>
				);
			};

			testComponent = mount(<TestComponent />);
		});

		expect(testComponent.html()).toMatchSnapshot();
	});

	it('should match snapshot for list with an open item', () => {
		let testComponent;
		act(() => {
			const TestComponent = () => {
				return (
					<Expandable>
						<ExpandableItem
							open
							title={<h1>Title</h1>}
							body={<p>Body</p>}
						/>
					</Expandable>
				);
			};

			testComponent = mount(<TestComponent />);
		});

		expect(testComponent.html()).toMatchSnapshot();
	});

	it('should match snapshot for multi list with open and closed items', () => {
		let testComponent;
		act(() => {
			const TestComponent = () => {
				return (
					<Expandable>
						<ExpandableItem
							open
							title={<h1>Title1</h1>}
							body={<p>Body1</p>}
						/>
						<ExpandableItem
							open={false}
							title={<h1>Title2</h1>}
							body={<p>Body2</p>}
						/>
						<ExpandableItem
							open
							title={<h1>Title3</h1>}
							body={<p>Body3</p>}
						/>
					</Expandable>
				);
			};

			testComponent = mount(<TestComponent />);
		});

		expect(testComponent.html()).toMatchSnapshot();
	});

	it('should match snapshot for none-multi list with open and closed items', () => {
		let testComponent;
		act(() => {
			const TestComponent = () => {
				return (
					<Expandable multi={false}>
						<ExpandableItem
							open
							title={<h1>Title1</h1>}
							body={<p>Body1</p>}
						/>
						<ExpandableItem
							open={false}
							title={<h1>Title2</h1>}
							body={<p>Body2</p>}
						/>
						<ExpandableItem
							open
							title={<h1>Title3</h1>}
							body={<p>Body3</p>}
						/>
					</Expandable>
				);
			};

			testComponent = mount(<TestComponent />);
		});

		expect(testComponent.html()).toMatchSnapshot();
	});

	it('should pass on className to dom element', () => {
		const expandable = shallow(
			<Expandable className="expandable">
				<ExpandableItem className="expandable-item" />
				<ExpandableItem className="expandable-item" />
				<ExpandableItem className="expandable-item" />
				<ExpandableItem className="expandable-item" />
			</Expandable>,
		);
		expect(expandable.find('.expandable').length === 1).toBeTruthy();
		expect(expandable.find('.expandable-item').length === 4).toBeTruthy();
	});

	it('should set correct aria-expanded tag value', () => {
		let expandableItem;
		let spyedCallback;
		act(() => {
			const TestComponent = () => {
				const [open, setOpen] = useState(false);
				if (!spyedCallback)
					spyedCallback = jest.fn(() => setOpen(!open));

				return (
					<ExpandableItem
						open={open}
						className="expandable-item"
						onClick={spyedCallback}
					/>
				);
			};

			expandableItem = mount(<TestComponent />);
		});

		expect(
			expandableItem.find('article[aria-expanded="false"]').length === 1,
		).toBeTruthy();
		expect(
			expandableItem.find('article[aria-expanded="true"]').length === 0,
		).toBeTruthy();

		act(() => {
			expandableItem.find('article>header').simulate('click');
		});

		expandableItem.update();

		expect(
			expandableItem.find('article[aria-expanded="false"]').length === 0,
		).toBeTruthy();

		expect(
			expandableItem.find('article[aria-expanded="true"]').length === 1,
		).toBeTruthy();

		expect(spyedCallback).toHaveBeenCalledTimes(1);
	});

	it('should fire onClick when an item is clicked', () => {
		let expandables;
		let spyedCallback;
		act(() => {
			const TestComponent = () => {
				spyedCallback = jest.fn();

				return (
					<Expandable>
						<ExpandableItem
							className="expandable-item"
							onClick={spyedCallback}
						/>
					</Expandable>
				);
			};

			expandables = mount(<TestComponent />);
		});

		expect(expandables.find('article').hasClass('open')).not.toBeTruthy();

		act(() => {
			expandables.find('article>header').simulate('click');
		});

		expandables.update();

		expect(expandables.find('article').hasClass('open')).toBeTruthy();

		expect(spyedCallback).toHaveBeenCalledTimes(1);
	});

	it('should fire onChange on the list when an item is clicked', () => {
		let expandables;
		let spyedCallback;
		act(() => {
			const TestComponent = () => {
				spyedCallback = jest.fn();

				return (
					<Expandable onChange={spyedCallback}>
						<ExpandableItem className="expandable-item" />
					</Expandable>
				);
			};

			expandables = mount(<TestComponent />);
		});

		expect(expandables.find('article').hasClass('open')).not.toBeTruthy();

		act(() => {
			expandables.find('article>header').simulate('click');
		});

		expandables.update();

		expect(expandables.find('article').hasClass('open')).toBeTruthy();

		expect(spyedCallback).toHaveBeenCalledTimes(1);
	});

	it('should fire onChange on the list item when openStatus changes', () => {
		let expandableItem;
		let spyedCallback;
		act(() => {
			const TestComponent = () => {
				const [open, setOpen] = useState(false);
				if (!spyedCallback) spyedCallback = jest.fn();

				return (
					<ExpandableItem
						open={open}
						className="expandable-item"
						onChange={spyedCallback}
						onClick={() => setOpen(!open)}
					/>
				);
			};

			expandableItem = mount(<TestComponent />);
		});

		expect(spyedCallback).not.toHaveBeenCalled();

		act(() => {
			expandableItem.find('article>header').simulate('click');
		});

		expect(spyedCallback).toHaveBeenCalledTimes(1);
		expect(spyedCallback).toHaveBeenCalledWith(true);
		spyedCallback.mockReset();

		expect(spyedCallback).not.toHaveBeenCalled();

		act(() => {
			expandableItem.find('article>header').simulate('click');
		});

		expect(spyedCallback).toHaveBeenCalledTimes(1);
		expect(spyedCallback).toHaveBeenCalledWith(false);
	});

	it('should add correct gaps to opened list items', () => {
		let expandables;
		act(() => {
			const TestComponent = () => {
				return (
					<Expandable multi>
						<ExpandableItem className="i1" />
						<ExpandableItem className="i2" />
						<ExpandableItem className="i3" />
						<ExpandableItem className="i4" />
						<ExpandableItem className="i5" />
					</Expandable>
				);
			};

			expandables = mount(<TestComponent />);
		});

		expect(expandables.find('article.open').length === 0).toBeTruthy();

		expandables.find('article.i1>header').simulate('click'); // Open first item
		expandables.update();

		expect(expandables.find('article.i1').hasClass('open')).toBeTruthy();
		expect(
			expandables.find('article.i1').hasClass('topGap'),
		).not.toBeTruthy();
		expect(
			expandables.find('article.i1').hasClass('bottomGap'),
		).toBeTruthy();

		expandables.find('article.i2>header').simulate('click'); // Open second item
		expandables.update();

		expect(expandables.find('article.i1').hasClass('open')).toBeTruthy();
		expect(
			expandables.find('article.i1').hasClass('topGap'),
		).not.toBeTruthy();
		expect(
			expandables.find('article.i1').hasClass('bottomGap'),
		).toBeTruthy();

		expect(expandables.find('article.i2').hasClass('open')).toBeTruthy();
		expect(
			expandables.find('article.i2').hasClass('topGap'),
		).not.toBeTruthy();
		expect(
			expandables.find('article.i2').hasClass('bottomGap'),
		).toBeTruthy();

		expandables.find('article.i1>header').simulate('click'); // Close first item
		expandables.update();

		expect(
			expandables.find('article.i1').hasClass('open'),
		).not.toBeTruthy();
		expect(
			expandables.find('article.i1').hasClass('topGap'),
		).not.toBeTruthy();
		expect(
			expandables.find('article.i1').hasClass('bottomGap'),
		).not.toBeTruthy();

		expect(expandables.find('article.i2').hasClass('open')).toBeTruthy();
		expect(expandables.find('article.i2').hasClass('topGap')).toBeTruthy();
		expect(
			expandables.find('article.i2').hasClass('bottomGap'),
		).toBeTruthy();

		expandables.find('article.i5>header').simulate('click'); // Open last item
		expandables.update();

		expect(expandables.find('article.i5').hasClass('open')).toBeTruthy();
		expect(expandables.find('article.i5').hasClass('topGap')).toBeTruthy();
		expect(
			expandables.find('article.i5').hasClass('bottomGap'),
		).not.toBeTruthy();

		expandables.find('article.i4>header').simulate('click'); // Open one before last item
		expandables.update();

		expect(expandables.find('article.i4').hasClass('open')).toBeTruthy();
		expect(expandables.find('article.i4').hasClass('topGap')).toBeTruthy();
		expect(
			expandables.find('article.i4').hasClass('bottomGap'),
		).toBeTruthy();

		expect(expandables.find('article.i5').hasClass('open')).toBeTruthy();
		expect(
			expandables.find('article.i5').hasClass('topGap'),
		).not.toBeTruthy();
		expect(
			expandables.find('article.i5').hasClass('bottomGap'),
		).not.toBeTruthy();
	});
});
