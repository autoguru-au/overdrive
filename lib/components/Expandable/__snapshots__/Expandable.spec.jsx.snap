// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Expandable /> should match snapshot for list with a closed item 1`] = `"<div class=\\"expandable\\" aria-label=\\"expandable list\\"><article class=\\"expandableItem\\" aria-label=\\"expandable item\\" aria-expanded=\\"false\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title</h1></header></article></div>"`;

exports[`<Expandable /> should match snapshot for list with an open item 1`] = `"<div class=\\"expandable\\" aria-label=\\"expandable list\\"><article class=\\"expandableItem\\" aria-label=\\"expandable item\\" aria-expanded=\\"false\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title</h1></header></article></div>"`;

exports[`<Expandable /> should match snapshot for list without items 1`] = `"<div class=\\"expandable\\" aria-label=\\"expandable list\\"></div>"`;

exports[`<Expandable /> should match snapshot for multi list with open and closed items 1`] = `"<div class=\\"expandable\\" aria-label=\\"expandable list\\"><article class=\\"expandableItem\\" aria-label=\\"expandable item\\" aria-expanded=\\"false\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title1</h1></header></article><article class=\\"expandableItem\\" aria-label=\\"expandable item\\" aria-expanded=\\"false\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title2</h1></header></article><article class=\\"expandableItem open\\" aria-label=\\"expandable item\\" aria-expanded=\\"true\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title3</h1></header><section aria-label=\\"expandable item body\\" class=\\"body\\"><p>Body3</p></section></article></div>"`;

exports[`<Expandable /> should match snapshot for none-multi list with open and closed items 1`] = `"<div class=\\"expandable\\" aria-label=\\"expandable list\\"><article class=\\"expandableItem\\" aria-label=\\"expandable item\\" aria-expanded=\\"false\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title1</h1></header></article><article class=\\"expandableItem\\" aria-label=\\"expandable item\\" aria-expanded=\\"false\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title2</h1></header></article><article class=\\"expandableItem open\\" aria-label=\\"expandable item\\" aria-expanded=\\"true\\"><header aria-label=\\"expandable item header\\" class=\\"title\\"><i style=\\"width: 16px; height: 16px;\\" class=\\"root arrow\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path d=\\"M158.08 183.04L256 280.96l97.92-97.92L384 213.333l-128 128-128-128 30.08-30.293z\\"></path></svg></i><h1>Title3</h1></header><section aria-label=\\"expandable item body\\" class=\\"body\\"><p>Body3</p></section></article></div>"`;
